crates/runtime/src/datafusion.rs

```diff
@@ -1360,20 +1360,31 @@ impl DataFusion {
                 reason: format!("Failed to create view acceleration: {e}"),
             })?;
 
+        // Detect if data for view was already loaded so we don't need to wait for the first refresh to complete to mark it as ready.
+        let mut initial_load_complete = false;
+        if let Ok(checkpoint) = DatasetCheckpoint::try_new(view).await {
+            if checkpoint.exists().await {
+                initial_load_complete = true;
+            }
+        }
+
         // TODO Extend and refactor (move to helper functionality that can be re-used with datasets logic)
```

Comment:
Have a work item to complete this TODO - it seems there is a room to share the same logic when applying acceleration params when creating Dataset and View and refactor this.